app/controllers/dogs_controller.rb

```diff
@@ -4,7 +4,13 @@ class DogsController < ApplicationController
   # GET /dogs
   # GET /dogs.json
   def index
-    @dogs = Dog.all
+    # set default page to 0 for root page
+    params["page"] == nil ? @page = 0 : @page = params["page"].to_i
+    # 5 dogs per page
+    dogs_per_page = 5
```

Comment:
Should set this as a constant outside the methods at the top of the file since you're using it as such.
`DOGS_PER_PAGE = 5`